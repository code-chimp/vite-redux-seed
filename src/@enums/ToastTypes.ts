enum ToastTypes {
  Error = 'danger',
  Info = 'info',
  Success = 'success',
  Warning = 'warning',
}

export default ToastTypes;
