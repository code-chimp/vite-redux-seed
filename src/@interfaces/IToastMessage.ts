import ToastType from '../@types/ToastType';

interface IToastMessage {
  id: string;
  type?: ToastType;
  text?: string;
}

export default IToastMessage;
