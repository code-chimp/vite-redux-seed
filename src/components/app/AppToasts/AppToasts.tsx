import React, { FC } from 'react';
import { useAppSelector } from '../../../helpers';
import { selectToasts } from '../../../store/slices/toasts';
import Toast from './Toast';

const AppToasts: FC = () => {
  const messages = useAppSelector(selectToasts);

  return (
    <div className="toast-container position-fixed bottom-0 end-0 p-3">
      {messages.map(message => (
        <Toast key={message.id} toastMessage={message} />
      ))}
    </div>
  );
};

export default AppToasts;
