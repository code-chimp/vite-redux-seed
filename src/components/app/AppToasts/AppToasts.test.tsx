import React from 'react';
import { render } from '@testing-library/react';
import configureMockStore, { MockStore } from 'redux-mock-store';
import { Provider } from 'react-redux';
import IToastMessage from '../../../@interfaces/IToastMessage';
import ToastTypes from '../../../@enums/ToastTypes';
import AppToasts from './AppToasts';

const mockStore = configureMockStore([]);
let store: MockStore;

const initialState: Array<IToastMessage> = [
  {
    id: 'foo',
    text: 'batten down the hatches',
    type: ToastTypes.Warning,
  },
];

describe('components / app / AppToasts', () => {
  it('should match the snapshot', () => {
    store = mockStore({ toasts: initialState });
    const { asFragment } = render(
      <Provider store={store}>
        <AppToasts />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
