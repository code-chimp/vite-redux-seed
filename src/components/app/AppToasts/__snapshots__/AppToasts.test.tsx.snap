// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components / app / AppToasts > should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="toast-container position-fixed bottom-0 end-0 p-3"
  >
    <div
      aria-atomic="true"
      aria-live="assertive"
      class="toast fade show showing"
      data-testid="toast-foo"
      role="alert"
    >
      <div
        class="toast-header  bg-opacity-25 bg-warning text-warning"
        data-testid="toast-foo-header"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-triangle-exclamation flex-shrink-0 me-2"
          data-icon="triangle-exclamation"
          data-prefix="fas"
          data-testid="toast-foo-icon"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7 .2 40.1S486.3 480 472 480H40c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8 .2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24V296c0 13.3 10.7 24 24 24s24-10.7 24-24V184c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0 -64 0 32 32 0 1 0 64 0z"
            fill="currentColor"
          />
        </svg>
        <strong
          class="me-auto"
          data-testid="toast-foo-header-text"
        >
          Warning
        </strong>
        <button
          aria-label="Close"
          class="btn-close"
          data-bs-dismiss="toast"
          data-testid="toast-foo-close"
          type="button"
        />
      </div>
      <div
        class="toast-body"
        data-testid="toast-foo-body"
      >
        batten down the hatches
      </div>
    </div>
  </div>
</DocumentFragment>
`;
